extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/CustomFilterGenerator.java

```diff
@@ -159,12 +175,25 @@ static String generateContainerRequestFilter(MethodInfo targetMethod, ClassOutpu
                         filterMethod.readInstanceField(delegateField, filterMethod.getThis()),
                         getRequestFilterResultHandles(targetMethod, declaringClassName, filterMethod,
                                 getRRReqCtxHandle(filterMethod, rrContainerReqCtxHandle)));
+                String methodName;
+                switch (returnType) {
+                    case UNI_VOID:
+                        methodName = "handleUniVoid";
+                        break;
+                    case UNI_RESPONSE:
+                        methodName = "handleUniResponse";
+                        break;
+                    case UNI_REST_RESPONSE:
+                        methodName = "handleUniRestResponse";
+                        break;
+                    default:
+                        throw new IllegalStateException("ReturnType: '" + returnType + "' is not supported here");
```

Comment:
will this exception be caught and rethrown with additional info on the place the error happened? If not, it would be good to include some info on the location of the invalid declaration

Reply:
Very good idea

Reply:
My comment is more from a user perspective - it's just confusing to see all this stuff for this API.

Reply:
Done.